frontend/src/store/FamLoginUserState.ts

```diff
@@ -207,7 +207,61 @@ const getCachedAppRolesForDelegatedAdmin = (
             ? -1
             : 1;
     });
-}
+};
+
+const getMyCachedPermissions = () => {
+    let myRoles: any = [];
+    getUserAccess()?.map((item) => {
+        if (item.auth_key === AdminRoleAuthGroup.FamAdmin) {
+            const famGrant = item.grants.find((grant) => {
+                return grant.application.name === FAM_APPLICATION_NAME;
+            });
+            myRoles.push({
+                adminRole: 'Admin',
+                application: famGrant?.application.description,
+                env: famGrant?.application.env,
+                role: 'Admin',
+            });
+        }
+
+        if (item.auth_key === AdminRoleAuthGroup.AppAdmin) {
+            item.grants.map((grant) => {
+                myRoles.push({
+                    adminRole: 'Admin',
+                    application: grant.application.description,
+                    env: grant.application.env,
+                });
+            });
+        }
+
+        if (item.auth_key === AdminRoleAuthGroup.DelegatedAdmin) {
+            item.grants.map((grant) => {
+                grant.roles?.forEach((role) => {
+                    if(!role.forest_clients) {
+                        myRoles.push({
+                            adminRole: 'Delegated Admin',
+                            application: grant.application.description,
+                            env: grant.application.env,
+                            clientId: null,
+                            role: role.name,
```

Comment:
oh I see what you're trying to do here, maybe let's just use `role: 'Delegated Admin, ' + role.name`

Reply:
good one! Thanks!